### DISCUSSION TITLE
How to refresh playlist cache?

### DISCUSSION THREAD

**Discussion by JulesAU** [Goldenfreddy0703/plugin.audio.ytmusic.exp#36] [Q&A] [answered]
Use case: I prepare my playlists on my laptop in Google Music. Later, I control my kodi instance via a remote (or webUI) to start playing music. I leave my TV off, so I can't go into YTMusic settings to force it to clear the cache.

Any way I can force this from the API or SSH into the kodi box? I tried emptying the records in the sqlite DB 'gpmusic.db', but whilst that removed the playlists/songs, it didn't trigger a refresh of the data from YTMusic API.

Even if you point me to the right part of the code I can probably add the functionality and submit a pull request.

Thanks!

**Comment by JulesAU** [comment]
@Goldenfreddy0703  sorry to bug you; not sure if this is the right place to ask a Q?

**Reply by Goldenfreddy0703** [reply]
Unfortunately im not too sure on that but in context menu, there is an update library option, maybe that could be or you may have to go into addon settings and go to the advanced. @fischcode may know the answer sense he has done a great job with this addon.

**Comment by fischcode** [comment]
You can update the entire library cache (including playlists) in the context menu of the YtMusic Library menu item and the playlist cache (only) in the context menu of the playlist menu item inside the YtMusic Library. 
Since context menus aren't available using a remote, I would like to develop a 'headless mode' with access via separate menus / menu items to all the goodies normally hidden in the context menus, but am still struggling with a proof of concept. Maybe I should just start with a feature that optionally allows updating the cache from a menu item instead of a context menu. Would that help you?

**Comment by JulesAU** [comment]
Thanks for the detailed reply; yep a menu item should work because it's available via remotes right?

**Comment by fischcode** [comment, marked as answer]
Please take a look at the latest release (1.0-beta28) and make sure you activate the 'headless mode' setting. You should find menu options to reload the entire library or playlists only.
